version: '3.3'
services:
  db:
    image: pgvector/pgvector:pg15
    restart: always
    ports:
      - '5432:5432'
    volumes:
       - ./etc/db-ext-vector-test.sql:/docker-entrypoint-initdb.d/init.sql
       - postgres_data:/var/lib/postgresql/data/
    environment:
      POSTGRES_USER: talkdai
      POSTGRES_PASSWORD: talkdai
      POSTGRES_DB: test_talkdai
      POSTGRES_EXTENSIONS: vector
    healthcheck:
      test: ["CMD", "pg_isready", "-d", "test_talkdai", "-U", "talkdai"]
      interval: 10s
      timeout: 5s
      retries: 5
  dialog:
    build:
      context: .
      dockerfile: Dockerfile
    # entrypoint: pytest -vvv
    stdin_open: true
    tty: true
    depends_on:
      db:
        condition: service_healthy
    volumes:
      - ./dialog_lib/:/app/dialog_lib/
      - ./samples/:/app/samples/
    environment:
      POSTGRES_USER: talkdai
      POSTGRES_PASSWORD: talkdai
      POSTGRES_DB: test_talkdai
      POSTGRES_HOST: db
      DATABASE_URL: postgresql://talkdai:talkdai@db:5432/test_talkdai


volumes:
  postgres_data:
